app/models/ext_management_system.rb

```diff
@@ -476,4 +481,16 @@ def stop_event_monitor_queue_on_credential_change
       self.stop_event_monitor_queue
     end
   end
+
+  def blacklisted_event_names
+    (
+      self.class.blacklisted_events(:enabled => true).pluck(:event_name) +
+      blacklisted_events.where(:enabled => true).pluck(:event_name)
+    ).uniq.sort
+  end
+
+  def self.blacklisted_events(options = {})
+    options.merge!(:provider_model => name, :ems_id => nil)
+    BlacklistedEvent.where(options)
+  end
```

Comment:
@lfu You do not need to pass options into the method.  The line `BlacklistedEvent.where(options)` will return an ActiveRecord_Relation object and the caller can append `where` conditions as needed.

With this change line 487 becomes `self.class.blacklisted_events.where(:enabled => true).pluck(:event_name)`